PNDA-4530: Refactored pnda_env_example.yaml to support openstack heat backend.

AbhilashS-MapleLabs opened the pull request:
PNDA-4530

Factored out the pnda_env to include sections specific to cloud infra type.
Note: This is the first part of integrating OpenStack Heat backend to "pnda-cli"  repo.

Validation carried to verify the functionality of AWS after changes:
1. RHEL pico and standard for HDP for Create.

AbhilashS-MapleLabs committed:
PNDA-4530: Refactored pnda_env_example.yaml to support openstack heat backend.

jeclarke commented on pnda_env_example.yaml:
```diff
@@ -1,4 +1,22 @@
-ec2_access:
+cloud_infrastructure :
```
I think this should be 'infrastructure' without the 'cloud' as existing-machines may not be cloud based.

janselva replied:
Sure James , will change it

jeclarke commented on pnda_env_example.yaml:
```diff
@@ -1,4 +1,22 @@
-ec2_access:
+cloud_infrastructure :
+  # cloud infrastructuer used for pnda deployment
```
typo infrastructuer

jeclarke commented on pnda_env_example.yaml:
```diff
@@ -45,6 +52,61 @@ cloud_formation_parameters:
+  publicProducerSubnetGateway: 10.0.2.1
+
+existing_machines_parameters:
+  #Below 2 variables used in package_install.sh for offline mode
```
package-install.sh uses ${vpcCidr}, I don't see privateSubnetCidr or publicProducerSubnetCidr in there.

jeclarke commented:
Also please rebase this carefully on develop, we removed support for ubuntu and this will need to build on top of that without regressing any of those changes. thanks!

jeclarke commented:
Please review the contribution guidelines: https://cwiki.pnda.io/display/public/PNDA/Submission+Guidelines

In particular:
"When ready, rebase your branch in interactive mode and squash all the commits to one commit"

AbhilashS-MapleLabs commented:
Sorry, it was a mistake, instead of checking-in clone branch got checked in this.
Will squash all commits into one when ready.

AbhilashS-MapleLabs commented:
Have rebased and addressed the request:

1. Changed the “cloud infrastructure” into infrastructure
2. Added new section in pnda_env for existing machine to support Offline mode : 
a. vpcCidr is applicable only for AWS because it’s under AWS section
b. for OpenStack and “existing machine” flavor, need to add PNDA and ingest network.

jeclarke commented:
Thanks for the updates, merged now.
